### ISSUE TITLE
Mir support in subnets

### ISSUE THREAD

**Issue by adlrocha** [consensus-shipyard/lotus#34]
Depends on #33.

## Background
While legacy-Eudico had support for Mir in subnets, for Spacenet, we integrated Mir as a first-class consensus that could be use by-design in Lotus. In order to be able to use Mir in subnets, it may require some slight modifications to our current integration approach. Mainly, we should add support for reconfiguration, and handle the validator set on-chain through the subnet-actor, so the subnet actor determines the validators that are allowed to propose new blocks in the subnet.

## Goal
The goal of this issue is to be able to deploy new subnets running Mir and with support for reconfiguration.

## Tasks
- [ ] #21 
- [ ] Use the subnet actor to determine the validator-set. This requires the implementation of https://github.com/consensus-shipyard/ipc-agent/issues/16
- [ ] _(general, not Mir-specific)_ Add a consensus check to ensure that only validators registered in the subnet-actors's validator set are allowed to propose new blocks.

**Comment by adlrocha** [comment]
This has been implemented and we are just pending an end-to-end test to see that the approach works. Closing in favor of https://github.com/consensus-shipyard/ipc-agent/issues/16 were this is being tracked.
